### ISSUE TITLE
Can't upload new files to a resource

### ISSUE THREAD

**Issue by jamy127** [hydroshare/hydroshare#3543]
I tried to add a text file to a resource with the following error. Please help fix this issue. Thanks! https://www.hydroshare.org/resource/c71e41fe816548889ee039e870098eb4/
![image](https://user-images.githubusercontent.com/3229860/63468741-d6515800-c425-11e9-8113-49a75339ef03.png)

**Comment by pkdash** [comment]
@sblack-usu I was able to reproduce this error on www. However, on beta I couldn't reproduce this.

**Comment by sblack-usu** [comment]
I verified beta and www are on the same commit for 1.25 so I don't know how you'd get different results in beta/www.  As for the error, there looks to be an unnecessary check whether you can upload a file to an aggregation folder that starts here in a receiver hook:
https://github.com/hydroshare/hydroshare/blob/develop/hs_composite_resource/receivers.py#L19

I believe this if block may be removed.
https://github.com/hydroshare/hydroshare/blob/develop/hs_composite_resource/models.py#L500

**Comment by pkdash** [comment]
I was able to reproduce only with the resource @jamy127 provided the link for. When I create a new resource I don't see this issue. I even created multiple netcdf aggregations and then tried to upload multiple txt files. It worked fine.

**Comment by jamy127** [comment]
Hi Pabitra,

How are you? Thanks for looking into this issue. I think this error
happened after I deleted some files and refreshed the page maybe before it
finished the delete action.

I report this because I had similar issues several times when I deleted
resources and it seems the resources crashed.

I feel this may happen to other users and impact their resources, so I
reported this issue.
If it is not necessary for fix, I am OK to delete the resource and close
the issue. Thank you: )

Sincerely,
Tian Gan

Pabitra Dash <notifications@github.com>于2019年8月22日 周四09:44写道：

**Comment by pkdash** [comment]
@jamy127 Don't close this issue now. Are you able to reproduce this issue?

**Comment by jamy127** [comment]
@pkdash I actually don’t know what specifically caused this error. I feel it is more like when a delete request is sent and the system may not be able to handle that if a user quickly refreshes the page or does other actions in the system.

**Comment by sblack-usu** [comment]
I agree we should leave this open.  I did a quick check of the resource in the database and didn't see any hanging files/aggregations.  We definitely need to investigate more and I think the check-resource management commands would be helpful in diagnosing once we get them updated.  

I do think we can fix the side effect you're seeing (not able to upload a file due to the validationerror) by removing the block of code I pointed out above.

**Comment by dtarb** [comment]
While we do not know how to recreate the exact steps that got the resource into this state, we know it is now in an inconsistent state that results in an inability to add files.  I suggest deploying an exact version of the www code, production database with this resource and the associated iRODS file on a testing server (e.g. beta) and then seeing what happens with debug messaging turned on.  This may help understand the state this resource is in and how to prevent it.

What is interesting is that a folder can be created in the resource and a file added within the new folder.

Other problems with this resource are
- An attempt to download the bag for this resource results in "Error.  Sorry an error occurred" no further explanation.
- An attempt to download zipped on one of the aggregations or any singe file in the resource results in "Resource bag cannot be created due to download poll errors: error".  This message is puzzling as zipping and downloading an aggregation should not create a bag.  It may just be a poorly worded message, but it may also be the code going some place unexpected.
- However download zipped works on the folder I created in the resource.

I was able to explore the above because in showing me the problem yesterday @gantian127 made me an owner of the resource.  I am happy to give anyone else permissions on this to explore what is going on, but moving the database and doing this on a debug platform with debug messages turned on may be more informative.

**Comment by pkdash** [comment]
@jamy127 , @dtarb , @sblack-usu  I was able to reproduce this issue. The happens if you delete any multi-file aggregation and then try to upload a file. Deleting a fileset doesn't lead to this issue. Can fix it this weekend.
